R/mod_trend.R

```diff
@@ -0,0 +1,166 @@
+#' trend UI Function
+#'
+#' @description A shiny Module.
+#'
+#' @param id,input,output,session Internal parameters for {shiny}.
+#'
+#' @noRd 
+#'
+#' @importFrom shiny NS tagList 
+mod_trend_ui <- function(id){
+  ns <- NS(id)
+  
+  tagList(
+    uiOutput(ns("dynamic_trendUI"))
+  )
+}
+
+#' trend Server Functions
+#'
+#' @noRd 
+mod_trend_server <- function(id, filter_data){
+  moduleServer( id, function(input, output, session){
+    ns <- session$ns
+    
+    memoised_comment_table <- memoise::memoise(comment_table, cache = session$cache) # create a session-level cacheable version of comment_table()
+    
+    # Super UI ----
+    output$dynamic_trendUI <- renderUI({
+      # Only show module contents if the data from the database is not empty 
+      validate(
+        need(
+          data_exists <- filter_data()$filter_data %>%
```

Comment:
I'm pretty sure this can be replaced with just `filter_data()$filter_data`. `need()` doesn't just take booleans, any non empty value is fine, see `?need`